src/modules/core/components/SingleUserPicker/SingleNFTPicker.tsx

```diff
@@ -12,15 +18,34 @@ interface Props extends ComponentProps<typeof SingleUserPicker> {
 const displayName = 'SingleUserPicker.SingleNFTPicker';
 
 const SingleNFTPicker = ({ data, ...props }: Props) => {
+  const renderAvatar = (_, item: SelectedNFT) => {
+    const selectedNFTData = getSelectedNFTData(item, data);
+    return (
+      <Avatar
+        size="xs"
+        placeholderIcon="nft-icon"
+        title="NFT"
+        avatarURL={selectedNFTData?.imageUri || undefined}
+        notSet={!selectedNFTData?.imageUri}
+      />
+    );
+  };
+
   const formattedData = useMemo(
     () =>
-      data.map((item) => ({
-        id: `${item.address} ${item.id}`,
-        profile: {
-          displayName: `${item.name || item.tokenName} #${item.id}`,
-          walletAddress: item.address,
-        },
-      })),
+      data.map((item) => {
+        const tokenName = item.name || item.tokenName;
+        const nftDisplayName = nftNameContainsTokenId(tokenName)
```

Comment:
Prevents duplication of id if it happens to be included in the title (as with the Neymar nft)